pytensor/tensor/rewriting/linalg.py

```diff
@@ -539,3 +539,21 @@ def svd_uv_merge(fgraph, node):
                     or len(fgraph.clients[cl.outputs[2]]) > 0
                 ):
                     return [cl.outputs[1]]
+
+
+@register_canonicalize
+@register_stabilize
+@node_rewriter([Blockwise])
+def rewrite_inv_inv(fgraph, node):
+    valid_inverses = (MatrixInverse, MatrixPinv)
```

Comment:
This also needs to include `Solve` and `SolveTriangular`, but you will need to check that the 2nd argument is `Eye`